dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,369 @@
+#include "dali/operators/reader/loader/webdataset_loader.h"
+#include <cstddef>
+#include <cstring>
+#include "dali/util/file.h"
+
+namespace dali {
+
+const std::set<DALIDataType> WebdatasetLoader::kSupportedTypes = {
+    DALI_UINT8, DALI_UINT16, DALI_UINT32,  DALI_UINT64, DALI_INT8,   DALI_INT16,
+    DALI_INT32, DALI_INT64,  DALI_FLOAT16, DALI_FLOAT,  DALI_FLOAT64};
+
+inline std::string SupportedTypesListGen() {
+  std::stringstream out;
+  for (auto& component_dtype : WebdatasetLoader::kSupportedTypes) {
+    out << component_dtype << ',';
+  }
+  if (!WebdatasetLoader::kSupportedTypes.empty()) {
+    out.unget();
+  }
+  return out.str();
+}
+
+inline WebdatasetLoader::MissingExt WebdatasetLoader::Str2MissingExt(
+    std::string missing_component_behavior) {
+  std::remove(missing_component_behavior.begin(), missing_component_behavior.end(), '_');
+  std::transform(missing_component_behavior.begin(), missing_component_behavior.end(),
+                 missing_component_behavior.begin(), static_cast<int (*)(int)>(std::tolower));
+  if (missing_component_behavior == "") {
+    return MissingExt::Empty;
+  } else if (missing_component_behavior == "skip") {
+    return MissingExt::Skip;
+  } else if (missing_component_behavior == "fillempty") {
+    return MissingExt::Empty;
+  } else if (missing_component_behavior == "raise") {
+    return MissingExt::Raise;
+  } else if (missing_component_behavior == "raiseerror") {
+    return MissingExt::Raise;
+  } else {
+    return MissingExt::Invalid;
+  }
+}
+
+WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
+    : Loader(spec),
+      uris_(spec.GetRepeatedArgument<std::string>("uris")),
+      configs_(spec.GetRepeatedArgument<std::string>("configs")),
+      missing_component_behavior_(
+          Str2MissingExt(spec.GetArgument<std::string>("missing_component_behavior"))) {
+  dtype_ = spec.HasArgument("dtype") ? spec.GetRepeatedArgument<DALIDataType>("dtype") :
+                                       std::vector<DALIDataType>(uris_.size(), DALI_UINT8);
+
+  DALI_ENFORCE(uris_.size() == configs_.size(),
+               "Number of uris does not match the number of config files");
+  DALI_ENFORCE(uris_.size() == dtype_.size(), "Number of uris does not match the number of types");
```

Comment:
As I understand uris are index files. So why you need to specify dtype for each index file?

Reply:
yeah, that is a bug, should be ext